src/pages/networks/forms/NetworkForwardForm.tsx

```diff
@@ -0,0 +1,333 @@
+  networkName,
+  project,
+}) => {
+  const notify = useNotify();
+
+  const { data: network } = useQuery({
+    queryKey: [queryKeys.networks, networkName, project],
+    queryFn: () => fetchNetwork(networkName ?? "", project ?? ""),
+  });
+
+  const updateFormHeight = () => {
+    updateMaxHeight("form-contents", "p-bottom-controls");
+  };
+  useEffect(updateFormHeight, [notify.notification?.message]);
+  useEventListener("resize", updateFormHeight);
+
+  const addPort = () => {
+    void formik.setFieldValue("ports", [
+      ...formik.values.ports,
+      {
+        protocol: "tcp",
+      },
+    ]);
+
+    const name = `ports.${formik.values.ports.length}.listenPort`;
+    setTimeout(() => document.getElementById(name)?.focus(), 100);
+  };
+
+  return (
+    <Form className="form network-forwards-form" onSubmit={formik.handleSubmit}>
+      <Row className="form-contents">
+        <Col size={12}>
+          {/* hidden submit to enable enter key in inputs */}
+          <Input type="submit" hidden />
+          <Row className="p-form__group p-form-validation">
+            <NotificationRow />
+            <Notification severity="information" title="Network information">
+              Name: {network?.name}
+              <br />
+              {network?.config["ipv4.address"] && (
+                <>
+                  IPv4 subnet: {network?.config["ipv4.address"]}
+                  <br />
+                </>
+              )}
+              {network?.config["ipv6.address"] && (
+                <>IPv6 subnet: {network?.config["ipv6.address"]}</>
+              )}
+            </Notification>
+          </Row>
+          <Input
+            {...formik.getFieldProps("listenAddress")}
+            id="listenAddress"
+            type="text"
+            label="Listen address"
+            placeholder="Enter IP address"
+            autoFocus
+            required
+            stacked
+            disabled={isEdit}
+            help="Any address routed to LXD."
+            error={
+              formik.touched.listenAddress
+                ? formik.errors.listenAddress
+                : undefined
+            }
+          />
+          <Input
+            {...formik.getFieldProps("defaultTargetAddress")}
+            id="defaultTargetAddress"
+            type="text"
+            label="Default target address"
+            help={
+              <>
+                Fallback target for traffic that does not match a port specified
+                below.
+                <br />
+                Must be from the network <b>{network?.name}</b>.
+              </>
+            }
+            placeholder="Enter IP address"
+            stacked
+          />
+          <Input
+            {...formik.getFieldProps("description")}
+            id="description"
+            type="text"
+            label="Description"
+            placeholder="Enter description"
+            stacked
+          />
+          {formik.values.ports.length > 0 && (
```

Comment:
The add ports table makes this form component large and quite difficult to read from this point onwards. Can we create a `NetworkForwardPorts` table for this?